src/views/bouquets/BouquetEditView.vue

```diff
@@ -0,0 +1,147 @@
+<script setup>
+import { ref, onMounted } from "vue"
+import { useDatasetStore } from "../../store/DatasetStore"
+import { useBouquetStore } from "../../store/BouquetStore"
+import { useRouter, useRoute } from "vue-router"
+import SearchAPI from "../../services/api/SearchAPI"
+import Multiselect from "@vueform/multiselect"
+
+const searchAPI = new SearchAPI()
+const datasetStore = useDatasetStore()
+const bouquetStore = useBouquetStore()
+const router = useRouter()
+const route = useRoute()
+
+const isCreate = route.name === "bouquet_add"
+
+const form = ref({})
+const datasets = ref([])
+const selectedDataset = ref(null)
+const selector = ref(null)
+const loadedBouquet = ref({})
+
+const search = async (query) => {
+  if (!query) return []
+  const results = await searchAPI._search(query, { hitsPerPage: 10 })
+  return results.hits.map(r => {
+    return { value: r.id, label: r.title }
+  }).filter(r => !datasets.value.map(d => d.id).includes(r.value))
+}
+
+const onSelect = async (value) => {
+  const dataset = await datasetStore.load(value)
+  datasets.value.push(dataset)
+  selector.value.clear()
+  selector.value.refreshOptions()
+}
+
+const onDeleteDataset = (datasetId) => {
+  datasets.value = datasets.value.filter(d => d.id !== datasetId)
+}
+
+const onSubmit = async () => {
+  let res
+  const data = {
+    ...form.value,
+    datasets: datasets.value.map(d => d.id),
+  }
+  if (isCreate) {
+    res = await bouquetStore.create({
+      ...data,
+      tags: ["ecospheres"],
+      extras: { is_ecospheres: true },
+    })
+  } else {
+    res = await bouquetStore.update(loadedBouquet.value.id, {
+      ...data,
+      tags: loadedBouquet.value.tags,
+    })
```

Comment:
`extras` is preserved, right?

Reply:
yes, if we don't touch it it should be preserved.